### PR TITLE
chore(PPDSC-1581): upgrade to cypress 12

### PR DESCRIPTION
**Pull request by mstuartf**
[PPDSC-1581](https://nidigitalsolutions.jira.com/browse/PPDSC-1581)

**What**

1. Background - why this is needed
2. What did you do
- Renamed all files as required / updated extensions
- Moved config files to root dir
- Added handler to stop uncaught exceptions failing tests
- Updated setup for `test_newskit_in_nextjs_app` job
3. What does the reviewers should expect
- Existing cypress tests pass 

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

[PPDSC-1581]: https://nidigitalsolutions.jira.com/browse/PPDSC-1581?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-1581-cypress-upgrade
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-1581-cypress-upgrade/storybook

**Review by JohnTParsons — DISMISSED** [review]

**Review by JohnTParsons — DISMISSED** [review]

File: cypress.docs.config.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// cypress config must be in project root: https://github.com/cypress-io/cypress/issues/22689
+
+import {defineConfig} from 'cypress';
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
```
Comment (Xin00163) [inline cypress.docs.config.ts:7]: Not a big deal. Just checking the [docs](https://docs.cypress.io/guides/references/migration-guide#Config-Option-Changes), we can put the func `setupNodeEvents` in the plugin file

**Review by Xin00163 — APPROVED** [review]
Nice one! Do you think it's worth using the component testing feature or it's not needed for now?

File: cypress.components.config.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// cypress config must be in project root: https://github.com/cypress-io/cypress/issues/22689
+
+import {defineConfig} from 'cypress';
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
```
Comment (Xin00163) [inline cypress.components.config.ts:7]: would it be OK just to import the file `const setupNodeEvents = require('./cypress/plugins/index.js')`? https://docs.cypress.io/guides/references/migration-guide#Config-Option-Changes

File: cypress.docs.config.ts (hunk 1 above)
Comment (mstuartf) [reply cypress.docs.config.ts:7]: We do but you make a good point as we don't need it for any of the others. Will remove!

File: cypress.components.config.ts (hunk 2 above)
Comment (mstuartf) [reply cypress.components.config.ts:7]: Updated to match the intended new structure. Will just check the pipeline is ok then ready for another look. Thanks!

**Review by mutebg — APPROVED** [review]
another great one 👏

**Review by JohnTParsons — APPROVED** [review]

**Pull request merged**